### PR TITLE
feat(sfs): add a datasource to get the list of OBS targets

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a datasource to get the list of OBS targets.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource used to get list of OBS targets.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDataSourceSfsTurboObsTargets_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDataSourceSfsTurboObsTargets_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceSfsTurboObsTargets_basic
=== PAUSE TestAccDataSourceSfsTurboObsTargets_basic
=== CONT  TestAccDataSourceSfsTurboObsTargets_basic
--- PASS: TestAccDataSourceSfsTurboObsTargets_basic (843.73s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       843.797s
```

### CODE REVIEW

File: docs/data-sources/sfs_turbo_obs_targets.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `share_id` - (Required, String) Specifies the ID of the SFS Turbo file system to which the OBS target belongs.
+
+* `target_id` - (Optional, String) Specifies the ID of the OBS target.
+
+* `status` - (Optional, String) Specifies the status of the OBS target.
+  The valid values can be **AVAILABLE**, **MISCONFIGURED**, or **FAILED**.
```
Comment (Lance52259) [inline docs/data-sources/sfs_turbo_obs_targets.md]:
```suggestion
  The valid value can be **AVAILABLE**, **MISCONFIGURED** or **FAILED**.
```
or
```suggestion
  The valid values are **AVAILABLE**, **MISCONFIGURED** and **FAILED**.
```

File: huaweicloud/services/sfs/data_source_huaweicloud_sfs_turbo_obs_targets.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+func DataSourceSfsTurboObsTargets() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSfsTurboObsTargetsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the region in which to query the resource. If omitted, the provider-level region will be used.`,
```
Comment (Lance52259) [inline huaweicloud/services/sfs/data_source_huaweicloud_sfs_turbo_obs_targets.go]:
```suggestion
				Description: `Specifies the region in which to query the resource.`,
```

File: docs/data-sources/sfs_turbo_obs_targets.md (hunk 1 above)
Comment (ruwenqiang123) [reply docs/data-sources/sfs_turbo_obs_targets.md]: done

File: huaweicloud/services/sfs/data_source_huaweicloud_sfs_turbo_obs_targets.go (hunk 2 above)
Comment (ruwenqiang123) [reply huaweicloud/services/sfs/data_source_huaweicloud_sfs_turbo_obs_targets.go]: done

File: huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_obs_targets_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	})
+}
+
+func testAccDataSourceSfsTurboObsTargets_basic(name string, randInt int) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_sfs_turbo_obs_targets" "test" {
+  depends_on = [
+    huaweicloud_sfs_turbo_obs_target.test
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_obs_targets_test.go:59]: Here is no need to add `depends_on`.

**Pull request merged**